network/src/dms.rs

```diff
@@ -532,7 +526,9 @@ impl<N: GossipNetwork, S: Storage> DistributedMessageSet<N, S> {
     }
 
     /// Serves the gossip network node and the RPC server indefinitely, constantly updating the storage.
-    pub async fn serve(self) -> Result<Serve<Self, Error>, Error> {
+    ///
+    /// TODO: currently it just returns itself after some time.
+    pub async fn serve(self) -> Result<Self, Error> {
         let port_key = format!("dms-{}", self.read_state().await?.key);
```

Comment:
Suggets `let port_key = format!("dms-{}", self.read_state().await?.key);` -> `let port_key = self.get_key()` for all occurrences of `port_key`.

Reply:
This issue should be handled with a more sophisticated way. I created an issue (#228)

Reply:
ok